pkg/controllers/streaming/stream_controller.go

```diff
@@ -82,21 +95,244 @@ func (r *StreamReconciler) reconcile(ctx context.Context, logger logr.Logger, st
 	stream.Status.InitializeConditions()
 
 	// delegate to the provider via its REST API
-	logger.Info("calling provisioner for Stream", "provisioner", stream.Spec.Provider)
+	log.Info("calling provisioner for Stream", "provisioner", stream.Spec.Provider)
 	address, err := r.StreamProvisionerClient.ProvisionStream(stream)
 	if err != nil {
 		stream.Status.MarkStreamProvisionFailed(err.Error())
 		return ctrl.Result{Requeue: true}, err
 	}
 	stream.Status.MarkStreamProvisioned()
 	stream.Status.Address = *address
+
+	// reconcile binding metadata
+	childBindingMetadata, err := r.reconcileChildBindingMetadata(ctx, log, stream)
+	if err != nil {
+		log.Error(err, "unable to reconcile child binding metadata ConfigMap", "stream", stream)
+		return ctrl.Result{}, err
+	}
+	stream.Status.Binding.MetadataRef = corev1.LocalObjectReference{}
+	if childBindingMetadata != nil {
+		stream.Status.Binding.MetadataRef.Name = childBindingMetadata.Name
+	}
+
+	// reconcile binding secret
+	childBindingSecret, err := r.reconcileChildBindingSecret(ctx, log, stream)
+	if err != nil {
+		log.Error(err, "unable to reconcile child binding secret Secret", "stream", stream)
+		return ctrl.Result{}, err
+	}
+	stream.Status.Binding.SecretRef = corev1.LocalObjectReference{}
+	if childBindingSecret != nil {
+		stream.Status.Binding.SecretRef.Name = childBindingSecret.Name
+	}
+
+	if childBindingMetadata == nil {
```

Comment:
if both are nil, only metadata will be mentioned in the status message (probably ok)

Reply:
When reconciliation fails, what's is typically reported is the first failure, and not every possible subsequent failure. We could split the condition into separate values for the ConfigMap and Secret, but that seemed like overkill.